lua/neo-tree/sources/common/file-items.lua

```diff
@@ -119,7 +119,8 @@ function create_item(context, path, _type)
     end
     if f.hide_by_pattern then
       for _, p in ipairs(f.hide_by_pattern) do
-        if string.find(name, p) then
+        local match = string.find(p, '%' .. utils.path_separator) and path or name
```

Comment:
I think the Windows issue can be solved with this:

```suggestion
        local separator_pattern = utils.is_windows and '\\' or '/'
        local match = string.find(p, separator_pattern) and path or name
```